test/smoke/harness/graphql.go

```diff
@@ -91,43 +96,55 @@ func (h *Harness) GraphQLQueryT(t *testing.T, query string) *QLResponse {
 // handling authentication. Queries are performed with the provided UserID.
 func (h *Harness) GraphQLQueryUserT(t *testing.T, userID, query string) *QLResponse {
 	t.Helper()
+	retry := 1
+	var err error
+	var resp *http.Response
+	var tok string
 
 	h.mx.Lock()
-	tok := h.gqlSessions[userID]
+	tok = h.gqlSessions[userID]
 	if tok == "" {
 		tok = h.insertGraphQLUser(userID)
 	}
 	h.mx.Unlock()
 
-	query = strings.Replace(query, "\t", "", -1)
-	q := struct{ Query string }{Query: query}
+	for {
+		query = strings.Replace(query, "\t", "", -1)
+		q := struct{ Query string }{Query: query}
 
-	data, err := json.Marshal(q)
-	if err != nil {
-		h.t.Fatal("failed to marshal graphql query")
-	}
-	t.Log("Query:", query)
+		data, err := json.Marshal(q)
+		if err != nil {
+			h.t.Fatal("failed to marshal graphql query")
+		}
+		t.Log("Query:", query)
 
-	url := h.URL() + "/api/graphql"
-	req, err := http.NewRequest("POST", url, bytes.NewBuffer(data))
-	if err != nil {
-		t.Fatal("failed to make request:", err)
-	}
-	req.Header.Set("Content-Type", "application/json")
-	req.AddCookie(&http.Cookie{
-		Name:  auth.CookieName,
-		Value: tok,
-	})
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		t.Fatal("failed to make http request:", err)
+		url := h.URL() + "/api/graphql"
+		req, err := http.NewRequest("POST", url, bytes.NewBuffer(data))
+		if err != nil {
+			t.Fatal("failed to make request:", err)
+		}
+		req.Header.Set("Content-Type", "application/json")
+		req.AddCookie(&http.Cookie{
+			Name:  auth.CookieName,
+			Value: tok,
+		})
+		resp, err = http.DefaultClient.Do(req)
+		if err != nil {
+			t.Fatal("failed to make http request:", err)
+		}
+		if resp.StatusCode == 401 && retry > 0 {
+			h.mx.Lock()
+			tok = h.insertGraphQLUser(userID)
```

Comment:
As mentioned above, in this case we already know the user was created, so we can just call a method that creates a new session